src/main/java/org/tugraz/sysds/runtime/data/TensorBlock.java

```diff
@@ -290,25 +293,93 @@ public double get(int[] ix) {
 			return _denseBlock.get(ix);
 		}
 	}
-	
+
+	public String getString(int[] ix) {
+		if (_sparse) {
+			// TODO: Implement sparse
+			throw new NotImplementedException();
+			//return _sparseBlock.get(ix);
+		} else {
+			return _denseBlock.getString(ix);
+		}
+	}
+
 	public void set(int[] ix, double v) {
 		if (_sparse) {
 			throw new NotImplementedException();
 		} else {
 			_denseBlock.set(ix, v);
 		}
 	}
-	
+
+	public void set(int[] ix, String v) {
+		if (_sparse) {
+			throw new NotImplementedException();
+		} else {
+			_denseBlock.set(ix, v);
+		}
+	}
+
+	public void set(double v) {
+		if (_sparse) {
+			throw new NotImplementedException();
+		} else {
+			_denseBlock.set(v);
+		}
+	}
+
+	public void set(TensorBlock other) {
+		if (_sparse) {
+			throw new NotImplementedException();
+		} else {
+			if (other.isSparse()) {
+				throw new NotImplementedException();
+			} else {
+				_denseBlock.set(0, _dims[0], 0, _denseBlock.getCumODims(0), other.getDenseBlock());
+			}
+		}
+	}
+
+	public void set(MatrixBlock other) {
+		if (_sparse) {
+			throw new NotImplementedException();
+		} else {
+			if (other.isInSparseFormat()) {
+				throw new NotImplementedException();
+			} else {
+				_denseBlock.set(0, _dims[0], 0, _denseBlock.getCumODims(0), other.getDenseBlock());
+			}
+		}
+	}
+
+	public double sum() {
+		// TODO generalize this method to an aggregate method that can do more than just sum
+		if (_sparse) {
+			// TODO implement for sparse
+			throw new NotImplementedException();
+		} else {
+			double sum = 0;
+			int elementsPerRow = Arrays.stream(_dims, 1, _dims.length).reduce(1, (a, b) -> a * b);
+			for (int bix = 0; bix < _denseBlock.numBlocks(); bix++) {
+				double[] values = _denseBlock.valuesAt(bix);
+				for (int r = 0; r < _denseBlock.blockSize(bix); r++) {
+					int finalR = r;
+					sum += IntStream.range(0, elementsPerRow).mapToDouble(i -> values[finalR * elementsPerRow + i]).sum();
+				}
```

Comment:
As we want to use this prototype for benchmarking, make a TODO perf to remind us that we need to replace the use of streams here and make it multi-threaded as otherwise we'll not be able to saturate peak memory bandwidth.